hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java

```diff
@@ -257,17 +257,18 @@ private void promoteTmpToTarget(Path tmpTarget, Path target, FileSystem fs)
     }
   }
 
-  private Path getTempFile(Path target, Mapper.Context context) {
+  private Path getTempFile(Path target, Mapper.Context context, FileSystem fileSystem) {
     Path targetWorkPath = new Path(context.getConfiguration().
         get(DistCpConstants.CONF_LABEL_TARGET_WORK_PATH));
 
     Path root = target.equals(targetWorkPath) ? targetWorkPath.getParent()
         : targetWorkPath;
-    Path tempFile = new Path(root, ".distcp.tmp." +
+    String tempFilePrefix = DistCpUtils.getTargetTempFilePrefix(fileSystem);
```

Comment:
assuming the Path is resolved, the FS Scheme is that of Path.toURI.getScheme(); no need to pass the FS around.